### PR TITLE
Execute With Timeout and enhanced clean up

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This pull request enhances the clean-up and execution workflow for cron jobs. A new method is added to exec: `ExecuteWithTimeout(timeout time.Duration)`, this method wraps the original `Execute` method with a timeout that will stop the execution if `timeout` is reached. The clean-up step is enhanced by adding the [flag](https://www.terraform.io/docs/cli/commands/init.html#lock-timeout-lt-duration-gt-) `-lock-timeout=<duration>` to Terraform's destroy command. This flag allows us to wait for terraform state's lock to be unlocked before attempting to destroy the resource, this tweak is useful in the event where we have to destroy our Equinix Metal instance even though its provision is not complete.

### CODE REVIEW

File: go/exec/exec.go [hunk 1]
```diff
@@ -193,6 +195,23 @@ func (e *Exec) Prepare() error {
 	return nil
 }
 
+// ExecuteWithTimeout will call execution's Execute method with the given timeout.
+// Note that a timeout too small will result in
```
Comment (GuptaManan100) [inline go/exec/exec.go:199]: incomplete comment

File: go/infra/equinix/equinix.go [hunk 2]
```diff
@@ -87,7 +87,7 @@ func (e *Equinix) CleanUp() error {
 	for _, varValue := range e.TerraformVarArray() {
 		destroyOpts = append(destroyOpts, varValue)
 	}
-
+	destroyOpts = append(destroyOpts, tfexec.LockTimeout("120s"))
```
Comment (GuptaManan100) [inline go/infra/equinix/equinix.go:90]: what is 120s over here

File: go/exec/exec.go [hunk 3]
```diff
@@ -193,6 +195,23 @@ func (e *Exec) Prepare() error {
...
+
+	go func() {
+		errs <- e.Execute()
+	}()
+
+	select {
+	case err := <-errs:
+		return err
+	case <-time.After(timeout):
+		return errors.New(ErrorExecutionTimeout)
```
Comment (GuptaManan100) [inline go/exec/exec.go:211]: should we not cleanup e over here. Even if it timedout, the server needs to be closed

Comment (frouioui) [reply go/exec/exec.go:211]: We let the caller of `ExecuteWithTimeout` deal with the error, as we do with `Execute`

File: go/exec/exec.go (hunk 1 above)
Comment (frouioui) [reply go/exec/exec.go]: Resolved via 2e32ccf

**Review by GuptaManan100 — APPROVED** [review]
:octocat:

**Pull request merged**